src/dynamics/ads_player/ads_player.js

```diff
@@ -222,13 +225,23 @@ Scoped.define("module:Ads.Dynamics.Player", [
                     }, this);
                     if (dynamics) {
                         dynamics.on("resize", function(dimensions) {
-                            // This part will listen to the resize even after adsManger will be destroyed
-                            if (this.adsManager && typeof this.adsManager.resize === "function") {
-                                this.adsManager.resize(
-                                    this.getAdWidth(),
-                                    this.getAdHeight(),
-                                    google.ima.ViewMode.NORMAL
-                                );
+                            const width = dimensions?.width || this.getAdWidth();
```

Comment:
Hi @EdmundsBankai I think will be better add condition to check if dimensions is Object type. In other case it can cause some trouble, as dynamics is more higher order component, and other components can trigger it without additional dimension argument which will throw error here on if ads will play.

Reply:
@sambua I decided to just use `this.getAdWidth()` and `this.getAdHeight()` and removed the dimensions object + dimensions check. No need to use the dimensions object (and handle cases where dimensions is not provided) if the getAdWidth and getAdHeight methods work just fine.